### PR TITLE
SPLICE-1728 Fix Cache Miss

### PR DESCRIPTION
**Pull request by jleach4**
DIctionary scans go crazy during DDL.

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryCache.java [hunk 1]
```diff
@@ -270,7 +270,7 @@ public Conglomerate conglomerateCacheFind(Long conglomId) throws StandardExcepti
     }
 
     public void conglomerateCacheAdd(Long conglomId, Conglomerate conglomerate,TransactionController xactMgr) throws StandardException {
-        if (!dd.canUseCache(xactMgr))
+        if (!dd.canUseCache(xactMgr)  && conglomId>=DataDictionary.FIRST_USER_TABLE_NUMBER)
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryCache.java]: We can't rely on DataDictionary.FIRST_USER_TABLE_NUMBER to separate user from system conglomerates. When upgrading we sometimes add new system tables and that would make user tables fall inside the "system tables" category. I would add a method that checks a conglomId against the coreInfo and nonCoreInfo arrays.

Comment (jleach4) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryCache.java]: Ugh...  Not good.

Comment (jleach4) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryCache.java]: EnvUtils#getTableType has the same issue.

**Comment by jleach4** [comment]
Closing...  Opening another pull request with a different approach.

**Pull request closed without merging**